master/usage/encrypt-comms.md

```diff
@@ -0,0 +1,125 @@
+  authority.
+
+- Configure each {{site.prodname}} component to verify the etcd server's identity and to present
+  a certificate to the etcd server that is signed by the etcd certificate authority.
+  - [{{site.nodecontainer}}](../reference/node/configuration)
+  - [`calicoctl`](./calicoctl/configure/etcd)
+  - [CNI plugin](../reference/cni-plugin/configuration#etcd-location) (Kubernetes and OpenShift only)
+  - [Felix](../reference/felix/configuration#etcdv3-datastore-configuration) (on [bare metal hosts](../getting-started/bare-metal/installation/))
+  - [Typha](../reference/typa/configuration#etcdv3-datastore-configuration) (often deployed in
+    larger Kubernetes deployments)
+  - [Neutron or ML2 driver](https://docs.projectcalico.org/master/getting-started/openstack/) (OpenStack only)
+  - [DHCP agent](https://docs.projectcalico.org/master/getting-started/openstack/) (OpenStack only)
+
+### Connections from {{site.prodname}} components to kube-apiserver (Kubernetes and OpenShift)
+
+We recommend enabling TLS on kube-apiserver, as well as the client certificate and JSON web token (JWT)
+authentication modules. This ensures that all of its communications with {{site.prodname}} components occur
+over TLS 1.2. The {{site.prodname}} components present either an X.509 certificate
+or a JWT to kube-apiserver so that kube-apiserver can verify their identities.
+
+### Connections from Felix to Typha (Kubernetes)
+
+The communication between Felix and Typha instances can be secured with TLS.
+This protects against:
+
+-  A rogue or impostor Typha sending incorrect endpoint or policy information
+   to the Felix instances that connect to it
+
+-  A rogue or impostor Felix wrongly receiving privileged information about the
+   overall cluster from the Typha that it connects to
+
+-  Any other process discovering privileged information about the overall
+   cluster by eavesdropping on the communication channel
+
+To use TLS, each Typha and Felix instance must have a certificate and key pair signed by
+a trusted CA. Typha and Felix then only accept TLS connections, and require each
+connecting client to present a certificate that is signed by a trusted CA and
+has an expected identity in its Common Name or URI SAN field.
+
+-  For a [SPIFFE](https://github.com/spiffe/spiffe)-compliant deployment you
+   should configure `ClientURISAN` (on Typha) and `TyphaURISAN` (on Felix) with an
+   [SPIFFE Identity](https://github.com/spiffe/spiffe/blob/master/standards/SPIFFE-ID.md#2-spiffe-identity).
+   In addition, you should provision Typha and Felix with client certificates
+   that have the same identity in their URI SAN field.
+
+-  Alternatively you can set `ClientCN` (on Typha) and `TyphaCN` (on Felix) to
+   values matching the `Common Name` in the Typha and Felix certificates.
+
+If you set both `ClientURISAN`/`TyphaURISAN` and `ClientCN`/`TyphaCN` parameters,
+client certificate only have to match either `ClientURISAN`/`TyphaURISAN` or
```

Comment:
I am not sure, but since this is a one-directional communication, I believe that one of these services is acting as the "server" in the exchange, and I think it is Felix in this case. Let me know if I am missing anything.

Reply:
The server in these connections is Typha.  Each Felix initiates connection to Typha.

However, certificate validation is symmetric: when a connection is initiated, Typha validates Felix's certificate, and Felix validates Typha's certificate.